solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpJdkSolrClient.java

```diff
@@ -173,8 +173,8 @@ public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
           "Timeout occurred while waiting response from server at: " + pReq.url, e);
     } catch (SolrException se) {
       throw se;
-    } catch (RuntimeException re) {
-      throw new SolrServerException(re);
+    } catch (IOException | RuntimeException e) {
```

Comment:
I found that `LBHttpSolrClient` needs a `SolrServerException` to detect a down node.  Yet we still can throw an `IOException` here during `prepareRequest`.  This roughly mimics the exception handling in `Http2SolrClient`.

Reply:
Okay; just needs a comment because it looks wrong

Reply:
Actually, maybe LBHttpSolrClient should be enhanced to catch the IOException to behave properly here?  Otherwise, it seems wrong just looking at this SolrClient to throw/not-throw based on some wrapping/delegating client unless we spell out these things in the documented semantics of this method.

Reply:
Indeed, the doc-comment on the `LBSolrClient` request method says,

>  If a request fails due to an IOException, the server is moved to the dead pool

So I restored the previous behavior in `HttpJdkSolrClient` and made the modification in `LBSolrClient`.  This new behavior is covered by `LBHttp2SolrClientIntegrationTest`.